### ISSUE TITLE
Implement new Agent Runs page

### ISSUE THREAD

**Issue by Pwuts** [Significant-Gravitas/AutoGPT#8780]
* Part of #8774
* Partially blocked by #9168

Runs will become a per-agent view.

### Sub-issues

- [ ] #8782
- [X] #8783
- [ ] #9181
- [ ] #9307
- [ ] #8784
- [X] #9182
- [ ] #9184
- [ ] #9183

### Designs

> \[!IMPORTANT\]
>
> The screenshots below are from a slightly older iteration of the designs. The following changes have been made:
>
> * The "Status" section (see screenshots 2, 3, 4) has been removed, may be implemented later
> * The "Share output" button needs to be defined further and will be implemented later
> * The "See run details" button will be implemented later
> * Responsive designs for other screen sizes have been added, see Figma

Finished & successful:

<img src="https://uploads.linear.app/a47946b5-12cd-4b3d-8822-df04c855879f/7e657acb-30a2-4b0b-b7aa-b4c670ce9692/51dd9d5d-70d7-42e9-9f43-a22038a5f7c4?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2E0Nzk0NmI1LTEyY2QtNGIzZC04ODIyLWRmMDRjODU1ODc5Zi83ZTY1N2FjYi0zMGEyLTRiMGItYjdhYS1iNGM2NzBjZTk2OTIvNTFkZDlkNWQtNzBkNy00MmU5LTlmNDMtYTIyMDM4YTVmN2M0IiwiaWF0IjoxNzMyNjQ0MDg3LCJleHAiOjMzMzAzMjA0MDg3fQ.V3e-xnIg-wbIGVaz_wzQth6AWVl7KlJtizjU8qNTANc " alt="image.png" width="2880" />

Running:

<img src="https://uploads.linear.app/a47946b5-12cd-4b3d-8822-df04c855879f/51855340-7fe7-4891-99da-f7553ae8f2b2/829b441f-9e82-4061-8d6d-a886a27eae6c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2E0Nzk0NmI1LTEyY2QtNGIzZC04ODIyLWRmMDRjODU1ODc5Zi81MTg1NTM0MC03ZmU3LTQ4OTEtOTlkYS1mNzU1M2FlOGYyYjIvODI5YjQ0MWYtOWU4Mi00MDYxLThkNmQtYTg4NmEyN2VhZTZjIiwiaWF0IjoxNzMyNjQ0MTYzLCJleHAiOjMzMzAzMjA0MTYzfQ.flrQ1ZM3CwDvL3B-aKZQmg1Dgzm2X-IxCRy9MD458W8 " alt="image.png" width="2880" />

Listening / scheduled:

<img src="https://uploads.linear.app/a47946b5-12cd-4b3d-8822-df04c855879f/412ad6f5-115f-46b8-9aa7-453f59850b6b/6436c47a-6f98-4f5a-b4b6-078bf7870a81?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2E0Nzk0NmI1LTEyY2QtNGIzZC04ODIyLWRmMDRjODU1ODc5Zi80MTJhZDZmNS0xMTVmLTQ2YjgtOWFhNy00NTNmNTk4NTBiNmIvNjQzNmM0N2EtNmY5OC00ZjVhLWI0YjYtMDc4YmY3ODcwYTgxIiwiaWF0IjoxNzMyNjQ0MjUzLCJleHAiOjMzMzAzMjA0MjUzfQ.76x6IRpVPKdGSKhVWdI-jClj9ZMH9pFMGRpMaENXnRk " alt="image.png" width="2880" />

Failed:

<img src="https://uploads.linear.app/a47946b5-12cd-4b3d-8822-df04c855879f/d01432df-4fb5-4259-bf37-e07c29328595/f628becb-6bee-49ac-b476-c15dadb1f78c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2E0Nzk0NmI1LTEyY2QtNGIzZC04ODIyLWRmMDRjODU1ODc5Zi9kMDE0MzJkZi00ZmI1LTQyNTktYmYzNy1lMDdjMjkzMjg1OTUvZjYyOGJlY2ItNmJlZS00OWFjLWI0NzYtYzE1ZGFkYjFmNzhjIiwiaWF0IjoxNzMyNjQ0MzEzLCJleHAiOjMzMzAzMjA0MzEzfQ.tvxEIlITNRQ7-tWTQTY7ohV_c74L9qlyP9SxmXqFwHw " alt="image.png" width="2880" />

### Highlights & notes

* Page path: `/agents/[id]` -> file path `src/app/agents/[id]/page.tsx`
* List of runs (redesigned)
  * Table -> list of cards
  * Create a pinned run template from an existing run **(new)**
  * Extra: filter/sort by status
* Run view
  * Show latest run by default, or a placeholder element "no runs yet"
  * Show block executions w/ details **(new)**
    * *I think we should iterate on the design for this view, because blocks can be used multiple times in a graph and can be executed multiple times during a run, and in the current design this isn't clearly communicated in the list of block outputs.*
  * Tab-scroller can be treated as a stretch goal
  * Action buttons - depend on run status

Data sources:

* All runs of the particular agent (real-time)
* When selected: specific execution status and outputs (real-time)
* For "New run": the agent's input schema
